### DISCUSSION TITLE
Export template for front ports

### DISCUSSION THREAD

**Discussion by robertmain** [netbox-community/netbox#13661] [General]
I feel like I'm missing something super obvious here, but I'm trying to create an export template for a device's front ports. Essentially, I'd like a way to turn a device's(specifically a patch panel) front ports into a CSV that I can then feed to my label maker. However, the issue I'm running into at the moment is that if I simply add `dcim > front ports` to the export template, there's no export option in the front ports list. 

![image](https://github.com/netbox-community/netbox/assets/537684/e1f2267a-5212-4207-8b45-11864532b20c)

In order to even see the export template, I have to add `dcim > device` which causes it to then show up in the dropdown menu under the "Export" button on the top right. That's fine, but clicking it causes NB to dump a query set that's basically just a list of all the devices (not unreasonable for that view, but also not exactly what I was after).

Is there something super obvious I'm missing here, or is what I'm trying to do not implemented/supported yet?

**Comment by candlerb** [comment]
Try navigating to `/dcim/front-ports/?device_id=NNN`

If you find this is something you need to do frequently, you can add a Custom Link (button) on the device page.

**Reply by robertmain** [reply]
Oh thank you! That fixed it. I didn't know that was a thing...

Incidentally, what difference does this option make?
![image](https://github.com/netbox-community/netbox/assets/537684/5b7fa200-65bc-4c61-85b7-86d7f34fadbd)

It seems to force a download whether it's checked or not.

**Reply by candlerb** [reply]
I presume it sets the Content-Disposition header. But the browser may still download anyway, e.g. if it doesn't understand the MIME type returned.

**Reply by candlerb** [reply]
> I didn't know that was a thing...

It's consistent: to get a list of devices you navigate to Devices > Devices, and to get a list of front ports you navigate to Devices > Front Ports. In both cases you can then filter the set and export it. Using the GUI, you can set a filter on front ports so that only front ports for a specific device are listed.

However, the lists of components you get as **tabs** on a device don't have the export buttons.
